### ISSUE TITLE
CV is undefined, even though the CVProvider console logs it as loaded.

### ISSUE THREAD

**Issue by medonni** [giacomocerquone/opencv-react#8]
Hi,

Been trying to implement this hook for a day or two now and i'm experiencing weird behaviour.

In my Component 1 i have

```
const ComponentOne = () => {
  const onLoaded = (cv) => {
    console.log("opencv loaded", cv);
  };
  return (
    <OpenCvProvider onLoad={onLoaded} openCvPath="/opencv/opencv.js">
      <Componenttwo />
    </OpenCvProvider>
  );
};

```

and in my Component 2 i have
```

import { useOpenCv } from "opencv-react";

const Testtwo = () => {
  const { loaded, cv } = useOpenCv();

  useEffect(() => {
    if (cv) {
      console.log("test");
      navigator.mediaDevices
        .getUserMedia({ video: true, audio: false })
        .then((stream) => {
          let video = videoRef.current;
          if (video) {
            video.srcObject = stream;
            video.load();
            video.play();

            let frame = new cv.Mat(video.height, video.width, cv.CV_8UC4);
            let cap = new cv.VideoCapture(video);

            // main(cap, frame);
          }
        })
        .catch(function (err) {
          console.log("An error occurred while loading stream! " + err);
        });
    }
  }, [cv]);
  return (
    <>
      <p>{loaded ? "Loaded openCV." : "Loading openCV..."}</p>
      <video id="videoInput" ref={videoRef}></video>{" "}
      <canvas id="canvasOutput"></canvas>
    </>
  );
```
};

The expected behaviour would be that CV loads in, stops being undefined and everything in the useEffect runs.
The actual behaviour is that the if(cv) never gets executed, since the CV variable stays undefined.

Console logging the CV shows it as undefined, but the CVProvider logs it correctly.

Removing the if(cv) check gives the following error
**TypeError: Cannot read properties of undefined (reading 'Mat')**

What could be the cause of this?

**Comment by giacomocerquone** [comment]
Makes sense!
The reason why this happens is that the `cv` value returned from the useOpenCv hook is a reference to the window.cv object that, when changed, doesn't cause a rerender and therefore it's undefined in your effect

Sadly the code you see right now in the codebase isn't aligned with what's published on npmjs otherwise this would be fixed. I'm about to publish it and then let me know if it's fixed

**Comment by giacomocerquone** [comment]
Just published version 0.0.6 https://www.npmjs.com/package/opencv-react
